### PR TITLE
add support for setting title via xterm escape seq to Terminal.app

### PR DESCRIPTION
**Pull request by onflapp**

### CODE REVIEW

File: Applications/Terminal/TerminalView.m [hunk 1]
```diff
@@ -2967,6 +2967,15 @@ - (void)setAlternateAsMeta:(BOOL)altAsMeta
 // ---
 // Title (window, icon)
 // ---
+- (NSString *)windowTitle
+{
+  return title_window;
+}
```
Comment (trunkmaster) [inline Applications/Terminal/TerminalView.m:2973]: I think it's better to name it `xtermTitle` to reflect its value and use it along with other similar methods: `-programPath`, `-deviceName` and `-windowSize` to construct title in TerminalWindow's `-updateTitleBar`.

File: Applications/Terminal/TerminalView.m [hunk 2]
```diff
@@ -2967,6 +2967,15 @@ - (void)setAlternateAsMeta:(BOOL)altAsMeta
...
+- (NSString *)miniWindowTitle
+{
+  return title_miniwindow;
+}
+
```
Comment (trunkmaster) [inline Applications/Terminal/TerminalView.m:2978]: This method is unused. Moreover I think miniwindow title should be constructed dirreferently then window title and its logic should be placed inside TerminalWindow class.

File: Applications/Terminal/Defaults.m [hunk 3]
```diff
@@ -426,6 +426,7 @@ - (void)setTerminalFont:(NSFont *)font
 const NSUInteger TitleBarDeviceName  = 1<<2;
 const NSUInteger TitleBarFileName    = 1<<3;
 const NSUInteger TitleBarWindowSize  = 1<<4;
+const NSUInteger TitleBarTermTitle   = 1<<5;
```
Comment (trunkmaster) [inline Applications/Terminal/Defaults.m:429]: I think `TitleBarXTermTitle` is more clear name.

File: Applications/Terminal/Preferences/English.lproj/TitleBar.gorm/data.classes [hunk 4]
```diff
@@ -17,6 +17,7 @@
 	    "showDefault:"
 	);
 	Outlets = (
+	    termTitleBtn,
```
Comment (trunkmaster) [inline Applications/Terminal/Preferences/English.lproj/TitleBar.gorm/data.classes:20]: I think `xtermTitleBtn` is more clear name.

**Comment by trunkmaster** [comment]
@onflapp do you want me to make requested changes?

**Comment by onflapp** [comment]
Absolutely, I was just running after something else.

**Pull request merged**